### PR TITLE
Allow TimescaleTuner to be initialized with arbitary number of timescales

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

This will be necessary when we have shape control so we don't have to type out a bunch of the same number in an input file.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/ControlSystem/TimescaleTuner.hpp [hunk 1]
```diff
@@ -45,14 +50,33 @@ class er;
  */
 
 class TimescaleTuner {
+ private:
+  struct TimescaleCreator {
```
Comment (wthrowe) [inline src/ControlSystem/TimescaleTuner.hpp]: Move this class to an anonymous namespace in the cpp file.

File: tests/Unit/ControlSystem/Test_Tags.cpp [hunk 2]
```diff
@@ -29,7 +31,46 @@ struct LabelA {};
 using system = control_system::TestHelpers::System<
     2, LabelA, control_system::TestHelpers::Measurement<LabelA>>;
 
-struct MetavarsEmpty {};
+struct MetavarsEmpty {
+  static constexpr size_t volume_dim = 3;
+};
+
+class FakeCreator : public DomainCreator<3> {
```
Comment (wthrowe) [inline tests/Unit/ControlSystem/Test_Tags.cpp]: Can this test use the `FakeCreator` class added in the previous commit?  They look the same except the other one has option-parsing stuff.

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Comment by knelli2** [comment]
Squashed suggested changes directly and rebased.

**Review by wthrowe — DISMISSED** [review]

**Comment by knelli2** [comment]
@wthrowe I realized I introduced a bug with these changes when I was testing it with the shape map. The `control_system::Tags::Controller` tag uses a timescale tuner when it is created from options, but that tuner wasn't initialized if the number of components hadn't been set yet. The fixup fixes this.

**Comment by wthrowe** [comment]
OK, squash.

**Review by wthrowe — APPROVED** [review]

**Pull request merged**